lib/components/deprecated-loading.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, ViewContainerRef, ComponentFactoryResolver, Input, ComponentFactory, TemplateRef, AfterViewInit } from '@angular/core'
+import { CommonModule } from '@angular/common';
+import { TreeModel } from '../models/tree.model';
+import { AdHocComponentFactoryCreator } from './adhoc-component-factory.service';
+
+@Component({
+  selector: 'LoadingComponent',
+  template: ''
+})
+export class LoadingComponent implements AfterViewInit {
+  @Input() loadingTemplate: TemplateRef<any>;
+
+  constructor(private treeModel: TreeModel,
+              private componentFactoryResolver: ComponentFactoryResolver,
+              private viewContainerRef: ViewContainerRef,
+              private adHocComponentFactoryCreator: AdHocComponentFactoryCreator) {
+  }
+
+  ngAfterViewInit() {
+    this._loadTreeNodeContent();
+  }
+
+  _loadTreeNodeContent() {
+    let componentFactory;
+    try {
+      componentFactory = this.componentFactoryResolver.resolveComponentFactory(this.treeModel.loadingComponent);
```

Comment:
I try componentFactoryResolver for the case where they have passed a component that is registered somewhere in their application.  If that fails, I know its an ad hoc component (created in tree-model.ts) that I need to instantiate with the adHocComponentFactoryCreator.